### ISSUE TITLE
Better support for reified types on literals.

### ISSUE THREAD

**Issue by mhevery** [angular/ts2dart#356]
Given:
```
var x = <{[k:string]:any}>{};
var y = <string[]>[];
```

We get:
```
var x = {} as Map<String, any>;
var y = [] as List<String>;
```

We want to get:
```
var x = <String, any>{};
var y = <String>[];
```

**Comment by mprobst** [comment]
We translate these as casts as they are casts in TS. Is there anything in particular this breaks?

You could also try:

```ts
let x: string[] = [];
```

**Comment by mhevery** [comment]
@mprobst Yes, it is a cast, but what I am trying to request is that when we are doing a cast in front of a type literals, then it is not treated as cast, but as reified type. So the current behavior is correct, but we would like to augment it.

BTW, in dart: `List<String> x = []` and `var x = <String>[]` are not the same as one is reified and the other is not. (@vsmenon am I correct?)

**Comment by mprobst** [comment]
Ah I see, the issue is that you need the type reification inside the container. I'll TAL.

**Comment by mprobst** [comment]
https://reviews.angular.io/D19

**Comment by mprobst** [comment]
Fixed, released in v0.7.33 and v0.8.9.

**Comment by mhevery** [comment]
Love it, thanks
